src/main/scala/regmapper/RegisterRouter.scala

```diff
@@ -0,0 +1,45 @@
+// See LICENSE.SiFive for license details.
+
+package freechips.rocketchip.regmapper
+
+import Chisel._
+import freechips.rocketchip.config.Parameters
+import freechips.rocketchip.diplomacy._
+import freechips.rocketchip.interrupts._
+import freechips.rocketchip.tilelink._
+
+case class RegisterRouterParams(
+  name: String,
+  compat: Seq[String],
+  base: BigInt,
+  size: BigInt = 4096,
+  concurrency: Int = 0,
+  beatBytes: Int = 4,
+  undefZero: Boolean = true,
+  executable: Boolean = false)
+
+abstract class RegisterRouter[T <: Data](val devParams: RegisterRouterParams, portBundle: => T)
+                                        (implicit p: Parameters)
```

Comment:
Why do you proxy all the devParams? Can the traits not reach devParams.undefZero/etc from the constructor val for some reason?